### ISSUE TITLE
Long Alert repeat interval(multiple weeks) realerts ever 5 days

### ISSUE THREAD

**Issue by s1sfa** [grafana/grafana#72308]
### What happened?

When I create an alert with a very high repeat interval(multiple weeks), I am getting realerts every 5 days.

This is very similar to the alertmanager defaults described in this issue: (the default data.retention is 5 days)
https://github.com/prometheus/alertmanager/issues/2934

In alertmanager this value can be increased using a command flag you can add when you run it. I am not able to find any options in Grafana to change this value.

### What did you expect to happen?

Alerts to repeat at the interval set, and not sooner. 

### Did this work before?

No

### How do we reproduce it?

1. Create an alert that will be matched with a notification with a repeat interval greater than 5 days
2. Alert will repeat at 5 days.

### Environment (with versions)?

Grafana version v9.5.1 (bc353e4b2d)

### Grafana platform?

A downloaded binary

**Comment by usmangt** [comment]
Hi @s1sfa,

Thanks for opening this issue.

Can you please provide some screenshots as how exactly you configured your alert for the query where you get repetitive alerts?

**Comment by s1sfa** [comment]
Email history of alert notifications: (It resolved temporarily on July 19th for some reason and resumed the every 5 day pattern from then)
<img width="53" alt="image" src="https://github.com/grafana/grafana/assets/13647528/cc2aebe4-6e38-42b3-a5bb-1bb65cc7d866">

Matching notification policy with repeat interval 999w. There are no instances in the alert state history that match any of the alerts that repeat on 5 day interval. 
<img width="488" alt="image" src="https://github.com/grafana/grafana/assets/13647528/b07dfd30-007e-4652-b180-89738386a6fb">

This issue doesn't seem super well documented in Alertmanager documentation itself. But if grafana is using Alertmanager it's probably the same as what was discussed here: https://github.com/prometheus/alertmanager/issues/2934 . Although, alertmanager has a command flag to workaround the issue. I can't find anything similar in Grafana to expand this beyond 5 days.

**Comment by ivosth** [comment]
Hi, I am also facing this issue and based on your comment I have had a look at the code and I think the error may be in the **retentionNotificationsAndSilences** parameter as it occurs with the issue in the prometheus repository https://github1s.com/grafana/grafana/blob/HEAD/pkg/services/ngalert/notifier/alertmanager.go#L38-L39

**Comment by grobinson-grafana** [comment]
Here is the issue to track this https://github.com/prometheus/alertmanager/issues/3473

**Comment by erlingpaulsen** [comment]
Is it possible to disable resending of alerts? It's a serious limitation if you are forced to accept resending of alerts every X hours, where X has to be more than 0 and less than 120.

**Comment by grobinson-grafana** [comment]
Yes, change [`notification_log_retention`](https://github.com/grafana/grafana/blob/v11.1.4/conf/defaults.ini#L1300C1-L1300C17) in the config file from `5d`.

**Comment by nicholasmedor** [comment]
Does the notification_log_retention Grafana setting actually affect Grafana's Alertmanager data.retention setting (and in effect, the maximum possible repeat interval for alerts)?  The notes in grafana.ini indicate that notification_log_retention is for how long Grafana keeps Alertmanager alert notification log entries and not how long Alertmanager stores alert state and history (which is how I understand the data.retention setting behaves for Alertmanager).  I've tried setting notification_log_retention to 10 days, but the maximum possible repeat interval is still 5 days.

**Comment by vincent-olivert-riera** [comment]
@grobinson-grafana , 

> Keep in mind that increasing this value will increase memory usage as Grafana keeps all notification entries in memory.

Is there any way to know how much memory is being used for that purpose?

Also, does that mean alert notifications will be forgotten (and therefore resent) if Grafana is restarted (due to a version update, for instance)?

**Comment by tlinnyc** [comment]
Was anyone ever able to find a workaround on this? Seems like even after setting notification_log_retention more than 5d, still doesn't work.

**Comment by nicholasmedor** [comment]
I had originally added notification_log_retention under the [alerting] section of the grafana.ini file which must have been incorrect. I moved it to the [unified_alerting] section, and the longer retention setting is now taking effect.

**Comment by rkaito94** [comment]
> I had originally added notification_log_retention under the [alerting] section of the grafana.ini file which must have been incorrect. I moved it to the [unified_alerting] section, and the longer retention setting is now taking effect.
Hi nicholasmedor. I have do the same way, added notification_log_retention under [unified_alerting] in grafana.ini but it doesn't work. Repeat interval still 5d :(.
does anyone have any solution :(
